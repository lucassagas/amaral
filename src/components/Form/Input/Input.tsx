import {
  FormControl,
  FormLabel,
  Input as ChakraInput,
  InputProps as ChakraInputProps,
  Text,
} from '@chakra-ui/react'
import { FieldValues, UseFormRegister } from 'react-hook-form'

interface InputProps extends ChakraInputProps {
  name: string
  label?: string
  required?: boolean
  errors?: { [x: string]: any }
  register: UseFormRegister<FieldValues>
}

export function Input({
  name,
  label,
  register,
  errors,
  required = false,
  ...rest
}: InputProps) {
  return (
    <FormControl>
      {label && (
        <FormLabel fontSize="sm" htmlFor="email">
          {label}
        </FormLabel>
      )}
      <ChakraInput
        id={name}
        focusBorderColor="orange.500"
        bgColor="gray.900"
        variant="filled"
        isInvalid={errors && errors[name]}
        _hover={{ bgColor: 'gray.900' }}
        size="md"
        {...register(name, { required })}
        {...rest}
      />
      {errors && errors[name] && (
        <Text fontSize="sm" color="red.500">
          Campo obrigat√≥rio
        </Text>
      )}
    </FormControl>
  )
}
