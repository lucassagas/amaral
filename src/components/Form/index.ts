export { FormLayout } from './FormLayout'
export { Input } from './Input'
export { Select } from './Select'
