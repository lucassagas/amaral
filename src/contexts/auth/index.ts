export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
