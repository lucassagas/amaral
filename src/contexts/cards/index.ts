export { CardContext, CardContextProvider } from './cards';

