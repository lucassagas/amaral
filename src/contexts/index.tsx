import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <QueryClientProvider client={queryClient}>
        {children}
        <ReactQueryDevtools />
      </QueryClientProvider>
    </ChakraProvider>
  )
}
