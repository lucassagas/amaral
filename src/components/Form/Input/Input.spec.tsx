import { Input } from '@/components/Form/Input'
import { render, screen } from '@testing-library/react'
import { AppProvider } from '../../../contexts/index'

const register = jest.fn()

describe('<Input/>', () => {
  it('should render input correctly', () => {
    const { container } = render(
      <AppProvider>
        <Input name="name" label="Nome" register={register} />
      </AppProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('should render error when input is required', () => {
    render(
      <AppProvider>
        <Input
          name="email"
          label="Nome"
          register={register}
          errors={{ email: {} }}
        />
      </AppProvider>
    )
    expect(screen.getByText(/campo obrigat√≥rio/i)).toBeInTheDocument()
  })
})
