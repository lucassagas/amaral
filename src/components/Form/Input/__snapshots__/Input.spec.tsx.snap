// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render input correctly 1`] = `
<div>
  <div
    class="chakra-form-control css-1kxonj9"
    role="group"
  >
    <label
      class="chakra-form__label css-pk5tju"
      for="email"
      id="field-1-label"
    >
      Nome
    </label>
    <input
      class="chakra-input css-uxk25m"
      id="name"
    />
  </div>
</div>
`;
